fdbclient/include/fdbclient/BlobGranuleCommon.h

```diff
@@ -113,6 +113,8 @@ struct BlobGranuleCipherKeysMeta {
 	}
 };
 
+// When updating this struct with new fields, you must update and add new api versioning for corresponding struct in
```

Comment:
I don't see how just updating the API versioning will help to deal with the changes of the structure. If they are likely to change we need a more abstract API for accessing them, consisting of a opaque pointer and a set of accessor functions.

Reply:
For an example, this is how apple is doing this with the get mapped range api:
https://github.com/apple/foundationdb/pull/8994
https://github.com/apple/foundationdb/pull/9257

It's clearly a suboptimal and complicated solution though.
A much more ideal solution would be to have these structs defined in a language-agnostic, evolvable format like protobuf/thrift/flatbuffers, such that it could be used in all of the bindings interchangeably, but that would be a very large project.

Reply:
For what it's worth, I think the get mapped range API has had its struggles with this too and I don't know if it began with a particular plan regarding API evolution. The opaque pointer and accessor functions is the traditional way that fdb_c has handled this problem, and if we are planning for some other strategy here it could be useful to describe it somewhere.

Reply:
Yeah.
I think for this API we know we are already making one new version in the release following this one that changes to an iterator model instead of a synchronous model for getting all of the blob granule descriptions, so it's likely the current iteration of the apis will be replaced anyway.
And for that release I'd in theory like to do something like I described above if it's practical, because I'm not particularly a fan of any of the existing options for evolving those structs.